test/e2e/etcdctl_test.go

```diff
@@ -0,0 +1,116 @@
+		{"etcdctl lease list", false, "", ""},
+		{"etcdctl lease revoke 694d74da6ca8020b", false, "", "etcdserver: requested lease not found"},
+		{"etcdctl lease timetolive 694d74da6ca8020b", false, "", ""},
+		// member
+		{"etcdctl member list", false, "", ""},
+		{"etcdctl member add tester", false, "", "member peer urls not provided"},
+		{"etcdctl member promote 8e9e05c52164694d", false, "", "etcdserver: member not found"},
+		{"etcdctl member remove 8e9e05c52164694d", false, "", "etcdserver: member not found"},
+		{"etcdctl member update 8e9e05c52164694d", false, "", "member peer urls not provided"},
+		{"etcdctl move-leader 8e9e05c52164694d", false, "", ""},
+		{"etcdctl put /etcdctl-check-perf/foo bar", false, "", ""},
+		{"etcdctl snapshot save ./backup.db", false, "", ""},
+		{"etcdctl snapshot status ./backup.db", false, "", ""},
+		{"etcdctl snapshot restore ./backup.db", false, "", ""},
+		{"etcdctl version", false, "", ""},
+		// skip
+		{"etcdctl auth enable", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl auth disable", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl lock", true, "distributed locks should not be done via etcdctl", ""},
+		{"etcdctl make-mirror", true, "make-mirror takes one destination argument", ""},
+		{"etcdctl migrate", true, "migrate from v2 is not support for OCP4", ""},
+		{"etcdctl role add", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl role delete", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl role get", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl role grant-permission", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl role list", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl role revoke-permission", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl user add", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl user delete", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl user get", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl user grant-role", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl user list", true, "k8s does not use internal etcd RBAC", ""},
+		{"etcdctl user revoke-role", true, "k8s does not use internal etcd RBAC", ""},
+		// broke
+		{"etcdctl check datascale", true, "TODO currently fails in test but works fine in container directly", ""},
+		{"etcdctl txn", true, "TODO figure out multiline command", ""},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s", tc.command), func(t *testing.T) {
+			if tc.skip {
+				t.Skip(tc.skipReason)
+			}
+
+			actual, err := exec.Command("oc", getOcExecArgs(tc.command, podName)...).CombinedOutput()
+			switch {
+			case err == nil && len(tc.expectedError) == 0:
+				//pass
+			case err == nil && len(tc.expectedError) != 0:
+				require.NoError(t, fmt.Errorf("expected error not reported: %q", tc.expectedError))
```

Comment:
Not familiar with this lib and it looks like you're introducing a parallel/competing assertion framework — I expected to see Ginkgo like all the other tests, e.g. `o.Expect(err).NotTo(o.HaveOccurred())`

Reply:
I see Ginko used in origin but I don't see it in operator e2e testing feels a little like overkill for these use cases but we can look into it.

Reply:
testify does have heavy usage in the org FTR https://github.com/search?q=org%3Aopenshift+testify&type=code